param/__init__.py

```diff
@@ -1230,14 +1257,20 @@ class Selector(SelectorBase):
 
     __slots__ = ['objects', 'compute_default_fn', 'check_on_set', 'names']
 
+    _slot_defaults = _dict_update(
+        SelectorBase._slot_defaults, objects=_update_selector_default,
+        compute_default_fn=None, check_on_set=_update_selector_check_on_set,
+        allow_None=None, instantiate=False, default=None,
+    )
+
     # Selector is usually used to allow selection from a list of
     # existing objects, therefore instantiate is False by default.
-    def __init__(self, objects=None, default=None, instantiate=False,
-                 compute_default_fn=None, check_on_set=None,
-                 allow_None=None, empty_default=False, **params):
+    def __init__(self, objects=Undefined, default=Undefined, instantiate=Undefined,
+                 compute_default_fn=Undefined, check_on_set=Undefined,
+                 allow_None=Undefined, empty_default=False, **params):
```

Comment:
Is there a reason `empty_default` doesn't inherit (with an argument default of `Undefined`)?

Reply:
Because `empty_default` is not a slot.